src/interface/pipe.rs

```diff
@@ -179,6 +351,9 @@ impl EmulatedPipe {
         let mut read_end = self.read_end.lock();
         let mut pipe_space = read_end.len();
         if nonblocking && (pipe_space == 0) {
```

Comment:
I think read with 0 length should always success even if the pipe space is 0. That's what I've read from Linux pipe implementation, and they checked length==0 at the beginning of both read_from_pipe and write_from_pipe.
And I think add a test testing write and read on 0 length might be good